### ISSUE TITLE
authData now defaults to {}, which causes an error in the Request module

### ISSUE THREAD

**Issue by tim-kos** [kvz/airbud#24]
Imagine an URL that is not protected by basic auth. Airbud now defaults the `auth` opt to `{}`, which passes the opt down to the Request object (because the value is not `null`). The request object then errors out:

```
>>> got error Error: no auth mechanism defined
    at Auth.onRequest (****node_modules/request/lib/auth.js:133:32)
    at Request.auth (****node_modules/request/request.js:1270:14)
    at Request.init (****node_modules/request/request.js:379:10)
    at new Request (****node_modules/request/request.js:128:8)
    at request (****node_modules/request/index.js:54:10)
    at Function.get (****node_modules/request/index.js:62:12)
    at Airbud._execute (****node_modules/airbud/lib/Airbud.js:183:22)
```

To workaround this one needs to explicitly set `auth: null` in the opts to `Airbud.json`.

Also disallowing "user:pass@" inside of URLs is a really breaking change.

**Comment by kvz** [comment]
Actually I don’t disallow it, I must have made a typo in the url and then wrongfully assumed request didn’t handle it. Likely we can just use 3.5.0 and use basic auth right in the url.

But also thanks for reporting this issue, should still be fixed!

**Comment by kvz** [comment]
Released as `+ airbud@3.5.3`
